### PR TITLE
API: Introduce '/api/v1/wallet_recovery/public_wrapping_key' endpoint

### PR DESCRIPTION
**Pull request by Riddlerrr**
task: https://www.pivotaltracker.com/story/show/176549183

### CODE REVIEW

File: app/controllers/api/v1/wallet_recovery_controller.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+class Api::V1::WalletRecoveryController < Api::V1::ApiController
+  # TODO: Which authorization use here?
```
Comment (Alexey1100) [inline app/controllers/api/v1/wallet_recovery_controller.rb:2]: I think regular set of authorisations:

- requires header
- requires auth
- auth by signature proof

File: app/controllers/api/v1/wallet_recovery_controller.rb [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+
+  def public_wrapping_key
+    private_key = ENV['WALLET_RECOVERY_WRAPPING_KEY'] || 'default_key'
+    public_key = Eth::Key.new(priv: private_key).public_key.key
```
Comment (Alexey1100) [inline app/controllers/api/v1/wallet_recovery_controller.rb:6]: I think we need to use `ed25519` here (ref `Comakery::APISignature`)

File: spec/controllers/api/v1/wallet_recovery_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
+require 'rails_helper'
+
+RSpec.describe Api::V1::WalletRecoveryController, type: :controller do
+  describe 'GET /public_wrapping_key' do
+    render_views
+
+    context 'with correct private key' do
+      # To check you can use https://gobittest.appspot.com/Address
+      it 'renders a successful response' do
+        ENV['WALLET_RECOVERY_WRAPPING_KEY'] = '18E14A7B6A307F426A94F8114701E7C8E774E7F9A47E2C2035DB29A206321725'
```
Comment (matjacdev515) [inline spec/controllers/api/v1/wallet_recovery_controller_spec.rb:10]: The tricky thing about this way of setting the value of ENV variable, that the value will not be rolled back on test finish. I saw the same things in different tests, maybe we need to consider using Rspec mocks for that instead

File: app/controllers/api/v1/wallet_recovery_controller.rb (hunk 2 above)
Comment (Riddlerrr) [reply app/controllers/api/v1/wallet_recovery_controller.rb:6]: In the task description mentioned `secp256k1`
@aquabu Could you confirm which crypto method we must use in this case?

File: spec/controllers/api/v1/wallet_recovery_controller_spec.rb (hunk 3 above)
Comment (Riddlerrr) [reply spec/controllers/api/v1/wallet_recovery_controller_spec.rb:10]: I agree it's bad to rewrite ANV vars in tests:
Changed it to:
```ruby
allow(ENV).to receive(:[]).with('WALLET_RECOVERY_WRAPPING_KEY').and_return('18E14A7B6A307F426A94F8114701E7C8E774E7F9A47E2C2035DB29A206321725')
```

File: app/controllers/api/v1/wallet_recovery_controller.rb (hunk 2 above)
Comment (Riddlerrr) [reply app/controllers/api/v1/wallet_recovery_controller.rb:6]: Ok, confirmed using `secp256k1` through Ethereum lib

**Review by aquabu — APPROVED** [review]

**Pull request merged**